### PR TITLE
test(co-term expansion): Admins should be able to purchase additional seats for existing license

### PR DESCRIPTION
**Pull request by yasserfaraazkhan**
This is initial positive flow for co-term expansion. 

<a href="https://gitpod.io/#https://github.com/mattermost/mattermost-test-management/pull/7"></a>

### CODE REVIEW

File: data/test-cases/cloud/customer-portal/co-term-expansion/Test-Case-1.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+**Step 1**
+Ref Story and Figma: [MM-47641](https://mattermost.atlassian.net/browse/MM-47641)
+
+Pre-requisites:
+
+- The workspace should be having an active License for X number of users
+- The workspace can be at the initial, mid or end of its license term
+- Only Admins should be able to add user to current plan (either Professional or Enterprice)
+- if the current license is ending in 60 or 30 days, the expansion option is not shown
```
Comment (AGMETEOR) [inline data/test-cases/cloud/customer-portal/co-term-expansion/Test-Case-1.md:46]: 60 days to be precise

File: data/test-cases/cloud/customer-portal/co-term-expansion/Test-Case-1.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+Test Steps:
+
+1. Login to portal. (or create a new account).
+2. Move to Lienses page. Verify current License details like Start (date), Expiry (date), Users count, Total (cost).
+3. If Active users are less than the Number for which license was purchased, Verify only "Download license" button is shown.
```
Comment (AGMETEOR) [inline data/test-cases/cloud/customer-portal/co-term-expansion/Test-Case-1.md:52]: Might not be possible to know this from the portal

File: data/test-cases/cloud/customer-portal/co-term-expansion/Test-Case-1.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+4. If Active users are more than the Number for which license was purchased, Verify "Purchase additional seats" & "Download license" buttons are shown.
+   Verify a CTA is displayed with color background to indicate the margin by with the user count has exceeded.
+   - Content of CTA with orange border "You have exceeded your license seat count. Purchase additional seats" (if it exceeds by 5 % )
+   - Content of CTA with Red border "You have exceeded your license seat count by X users. Purchase additional seats" (if it exceeds more than 10 % )
+5. Verify "cancel Auto-renew" option is displayed.
```
Comment (AGMETEOR) [inline data/test-cases/cloud/customer-portal/co-term-expansion/Test-Case-1.md:57]: We have not implemented this yet

File: data/test-cases/cloud/customer-portal/co-term-expansion/Test-Case-1.md [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+   - Verify the Cost calculation is pro-rated.
+9. Click on "Complete purchase" button
+10. Verify "updating workspace information" page is shown
+11. On successful purchase, Verify "Added X users more to your license" message is shown, with option buttons to "View Billing"
+12. Verify purchase details on "purchase history" page 
+
+**Expected**
+
+Verify the purchase for aditional users was successfull and purchase history page shows details of purchase.
+Details should include Date (date of purchase), Description (License name and annual/pro-rata subscription), Users, Price, Total, Status (purchase status)
```
Comment (AGMETEOR) [inline data/test-cases/cloud/customer-portal/co-term-expansion/Test-Case-1.md:70]: Unless this test will be connected to stripe, seeing the additional users after expansin might be difficult in cypress

**Comment by saturninoabril** [comment]
@yasserfaraazkhan I'm reducing the reviewers. Ideal would be 2 reviewers -- one from Dev team and the other from QA guild.

File: data/test-cases/cloud/customer-portal/co-term-expansion/Test-Case-1.md (hunk 4 above)
Comment (yasserfaraazkhan) [reply data/test-cases/cloud/customer-portal/co-term-expansion/Test-Case-1.md:70]: got it Allan , will update it

**Comment by yasserfaraazkhan** [comment]
> @yasserfaraazkhan I'm reducing the reviewers. Ideal would be 2 reviewers -- one from Dev team and the other from QA guild.

sure Saturn,

File: data/TEST_CASE_TEMPLATE.md [hunk 5]
```diff
@@ -1,7 +1,7 @@
 ---
 # (Required) Ensure all values are filled up
 name: "Change Test Title"
-status: Approved
+status: Draft
```
Comment (saturninoabril) [inline data/TEST_CASE_TEMPLATE.md:4]: Thanks! I'd suggest o default with `Active` since once approved, it should be used for testing.

Comment (yasserfaraazkhan) [reply data/TEST_CASE_TEMPLATE.md:4]: Noted!! :)

File: data/test-cases/cloud/customer-portal/co-term-expansion/Test-Case-1.md [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+8. Verify on RHS, License details like Name, "Additional seats" (count of users that exceeded) input field, Cost per user (count x months remaning in plan) "total cost", "complete purchase" button
+   - Verify the Cost calculation is pro-rated.
+9. Click on "Complete purchase" button
+10. Verify "updating workspace information" page is shown
+11. On successful purchase, Verify "Added X users more to your license" message is shown, with option buttons to "View Billing"
+12. Verify purchase details on "purchase history" page 
+
+**Expected**
+
+Verify the purchase for aditional users was successfull and purchase history page shows details of purchase.
```
Comment (saturninoabril) [inline data/test-cases/cloud/customer-portal/co-term-expansion/Test-Case-1.md:69]:
```suggestion
Verify the purchase for additional users was successful and purchase history page shows details of purchase.
```

File: data/test-cases/cloud/customer-portal/co-term-expansion/Test-Case-1.md [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+Pre-requisites:
+
+- The workspace should be having an active License for X number of users
+- The workspace can be at the initial, mid or end of its license term
+- Only Admins should be able to add user to current plan (either Professional or Enterprice)
+- if the current license is ending in 60 or 30 days, the expansion option is not shown
+
+Test Steps:
+
+1. Login to portal. (or create a new account).
```
Comment (saturninoabril) [inline data/test-cases/cloud/customer-portal/co-term-expansion/Test-Case-1.md:50]:
```suggestion
1. Login to portal or create a new account.
```

File: data/test-cases/cloud/customer-portal/co-term-expansion/Test-Case-1.md [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+2. Move to Lienses page. Verify current License details like Start (date), Expiry (date), Users count, Total (cost).
+3. If Active users are less than the Number for which license was purchased, Verify only "Download license" button is shown.
+4. If Active users are more than the Number for which license was purchased, Verify "Purchase additional seats" & "Download license" buttons are shown.
+   Verify a CTA is displayed with color background to indicate the margin by with the user count has exceeded.
+   - Content of CTA with orange border "You have exceeded your license seat count. Purchase additional seats" (if it exceeds by 5 % )
```
Comment (saturninoabril) [inline data/test-cases/cloud/customer-portal/co-term-expansion/Test-Case-1.md:55]:
```suggestion
   - Content of CTA with orange border "You have exceeded your license seat count. Purchase additional seats" (if it exceeds by 5% )
```

File: data/test-cases/cloud/customer-portal/co-term-expansion/Test-Case-1.md [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+   - Content of CTA with Red border "You have exceeded your license seat count by X users. Purchase additional seats" (if it exceeds more than 10 % )
```
Comment (saturninoabril) [inline data/test-cases/cloud/customer-portal/co-term-expansion/Test-Case-1.md:56]:
```suggestion
   - Content of CTA with Red border "You have exceeded your license seat count by X users. Purchase additional seats" (if it exceeds more than 10% )
```

**Review by saturninoabril — CHANGES_REQUESTED** [review]
Thanks @yasserfaraazkhan! LGTM, please see comments.

**Comment by saturninoabril** [comment]
@yasserfaraazkhan, in reference to [PULL_REQUEST_TEMPLATE.md](https://github.com/mattermost/mattermost-test-management/blob/main/.github/PULL_REQUEST_TEMPLATE.md):

1. Please update the PR title to follow [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/). For test case, it could be in the following format:
```
test(feature): general description of tests
```

or something like
```
test(co-term expansion): admins should be able to purchase additional seats for existing license
```

2. Also, is there a test server already up and running where we could try out and follow the test case in this PR?

**Comment by yasserfaraazkhan** [comment]
>  2. Also, is there a test server already up and running where we could try out and follow the test case in this PR?

I have implemented the suggestions. Thank you @AGMETEOR and @saturninoabril

@saturninoabril , the changes are in 2 PR separately. After the initial set-up changes go in, I'll write more scenario with hands-on UI.

**Comment by saturninoabril** [comment]
Removing LM for I'll represent the QA review.

**Comment by yasserfaraazkhan** [comment]
Hi @AGMETEOR @saturninoabril ,
I updated one condition and will be merging the PR.

**Pull request merged**